From 01d66f5790ae8771868672a8efd13ad35f722a81 Mon Sep 17 00:00:00 2001
From: Peter Zijlstra <a.p.zijlstra@chello.nl>
Date: Fri, 2 Sep 2011 14:29:33 +0200
Subject: [PATCH] printk: 'force_early_printk' boot param to help with
 debugging

Gives me an option to screw printk and actually see what the machine
says.

Signed-off-by: Peter Zijlstra <a.p.zijlstra@chello.nl>
Link: http://lkml.kernel.org/r/1314967289.1301.11.camel@twins
Signed-off-by: Thomas Gleixner <tglx@linutronix.de>
Link: http://lkml.kernel.org/n/tip-ykb97nsfmobq44xketrxs977@git.kernel.org

diff --git a/kernel/printk/printk.c b/kernel/printk/printk.c
index b52cb3f4439e..e9d789d40dbd 100644
--- a/kernel/printk/printk.c
+++ b/kernel/printk/printk.c
@@ -1513,6 +1513,13 @@ asmlinkage void early_printk(const char *fmt, ...)
  */
 static bool __read_mostly printk_killswitch;
 
+static int __init force_early_printk_setup(char *str)
+{
+	printk_killswitch = true;
+	return 0;
+}
+early_param("force_early_printk", force_early_printk_setup);
+
 void printk_kill(void)
 {
 	printk_killswitch = true;
-- 
2.1.2

